### DISCUSSION TITLE
Is there a way to turn off utoipa in production?

### DISCUSSION THREAD

**Discussion by timmyjose** [juhaku/utoipa#741] [General]
I've just started experimenting with `utoipa` (specifically with `utoipa-swagger-ui`), and it looks just like what I'm looking for. However, I couldn't find (did a cursory search of the docs) a way to turn it off in production. Is that something that `utoipa` supports out of the box, or would it be the (say) developer's responsibility to turn it off via routing (or some other strategy)? 

Thanks!

**Comment by juhaku** [comment]
>  Is that something that `utoipa` supports out of the box, or would it be the (say) developer's responsibility to turn it off via routing (or some other strategy)?

The `utoipa-swagger-ui` behind the scenes creates the router or the means to serve the Swagger UI and the underlying api doc via enpoint handler. There is no mechanism to "not to create" the enpoint handlers. Thus if such is needed it is entirely users responsibility to conditionally add the `SwaggerUi` type to the web application in same way you would otherwise conditionally register enpoint handlers.

**Comment by timmyjose** [comment]
@juhaku That makes sense. Thanks!
